### ISSUE TITLE
Crash When in a Detached HEAD State if VALIDATE_ALL_CODEBASE: false

### ISSUE THREAD

**Issue by Kurt-von-Laven** [nvuillam/mega-linter#850]
**Describe the bug**
Mega-Linter v4.46.0 crashes when run without an active branch unless run on the entire code base.

**To Reproduce**

1. Change directories into some small test repo.
1. `git checkout -b test`
1. `touch foo && git add foo && git commit -m "Add foo"`
1. `git rebase --interactive main`
1. Select to edit the commit you just made.
1. Start the rebase.
1. Run any flavor of Mega-Linter incrementally.
1. Mega-Linter crashes with the following stack trace:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/run.py", line 15, in <module>
    linter.run()
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/MegaLinter.py", line 105, in run
    self.collect_files()
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/MegaLinter.py", line 386, in collect_files
    all_files = self.list_files_git_diff()
  File "/usr/local/lib/python3.9/site-packages/megalinter-0.1-py3.9.egg/megalinter/MegaLinter.py", line 484, in list_files_git_diff
    current_branch = repo.active_branch.commit.hexsha
  File "/usr/local/lib/python3.9/site-packages/git/repo/base.py", line 797, in active_branch
    return self.head.reference
  File "/usr/local/lib/python3.9/site-packages/git/refs/symbolic.py", line 294, in _get_reference
    raise TypeError("%s is a detached symbolic reference as it points to %r" % (self, sha))
TypeError: HEAD is a detached symbolic reference as it points to 'ac801b2f7f704a54a2cb26f3af670523ebef3bdc'
[config] /src/.mega-linter.yaml + Environment variables
```

**Expected behavior**
I expected Mega-Linter to run successfully.

**Additional context**
It appears that Mega-Linter uses GitPython to get the active branch, but during a rebase, for example, there is no active branch since the repository is in a detached HEAD state.

**Comment by Kurt-von-Laven** [comment]
Now that #851 has merged, we get past this crash and crash slightly later when running in incremental mode locally.

**Comment by Kurt-von-Laven** [comment]
Fixed in v5.0.0.
